src/main/functions/DidJWT.ts

```diff
@@ -126,17 +117,12 @@ async function signDidJwtInternal(
   return await createDidJWT({ ...payload }, options, header);
 }
 
-async function signDidJwtExternal(
-  payload: AuthenticationRequestPayload | AuthenticationResponsePayload,
-  signatureUri: string,
-  authZToken: string,
-  kid?: string
-): Promise<string> {
+async function signDidJwtExternal(payload: IdTokenPayload, signatureUri: string, authZToken: string, kid?: string): Promise<string> {
   // todo: Create method. We are doing roughly the same multiple times
-  const alg = isEd25519DidKeyMethod(payload.did) || isEd25519DidKeyMethod(payload.iss) || isEd25519DidKeyMethod(kid) ? KeyAlgo.EDDSA : KeyAlgo.ES256K;
+  const alg = isEd25519DidKeyMethod(payload.sub) || isEd25519DidKeyMethod(payload.iss) || isEd25519DidKeyMethod(kid) ? KeyAlgo.EDDSA : KeyAlgo.ES256K;
 
   const body = {
-    issuer: payload.iss && payload.iss.includes('did:') ? payload.iss : payload.did,
+    issuer: payload.sub,
```

Comment:
I doubt this is correct. The alg is derived from either the sub or the iss. Now the issuer only assumes sub, which means there is a potential disconnect with the alg. 

You replaced did with sub. But probably over here you should have something similar to what was there before but then for sub and iss

Reply:
Fixed